altair/vegalite/v5/api.py

```diff
@@ -391,6 +397,20 @@ def check_fields_and_encodings(parameter: Parameter, field_name: str) -> bool:
 _DictOrSchema = Union[core.SchemaBase, typing.Dict[str, Any]]
 _StatementType = Union[core.SchemaBase, _DictOrStr]
 
+
+def _is_undefined(obj: Any) -> TypeIs[UndefinedType]:
```

Comment:
From the commit message:

> Direct copy from another PR https://github.com/dangotbanned/altair/blob/d607c70824860ef3478d7d3996d26dc516d69369/altair/vegalite/v5/api.py#L504

If I were to rewrite this now, I'd probably define it in `schemapi.py`

https://github.com/vega/altair/blob/9afd374a83f954e79a5fd6e4f8a8a9c4e5381c04/tools/schemapi/schemapi.py#L755-L758

I imagine it could be useful in other modules, but for simplicity I've just copied *as-is* from:
- #3427